luaui/configs/gridmenu_config.lua

```diff
@@ -23,175 +21,159 @@ local categories = {
-
-					if ugdef and ugdef.id and uCanBuild[ugdef.id] then
-						gridPosUnit[uid][r .. c] = ugdef.id
-						unitGridPos[uid][ugdef.id] = r .. c
+				local ugdefName  = ugrid[index]
+				if ugdefName then
+					local ugDef = UnitDefNames[ugdefName]
+					if ugDef and ugDef.id and uCanBuild[ugDef.id] then
+						gridPosUnit [uid][r .. c]  = ugDef.id
+						unitGridPos [uid][ugDef.id] = r .. c
 					end
 				end
 			end
 		end
 	end
 end
 
-
 for unitDefID, unitDef in pairs(UnitDefs) do
 	unitCategories[unitDefID] = categoryGroupMapping[unitDef.customParams.unitgroup] or BUILDCAT_UTILITY
 end
 
-
+---------------------------------------------------------------- small utilities
 local function getCategoryIndex(category)
-	if category == BUILDCAT_ECONOMY then return 1
-	elseif category == BUILDCAT_COMBAT then return 2
-	elseif category == BUILDCAT_UTILITY then return 3
-	elseif category == BUILDCAT_PRODUCTION then return 4
-	else return nil end
+	if     category == BUILDCAT_ECONOMY    then return 1
+	elseif category == BUILDCAT_COMBAT     then return 2
+	elseif category == BUILDCAT_UTILITY    then return 3
+	elseif category == BUILDCAT_PRODUCTION then return 4 end
 end
 
 local function constructBuildOption(uDefID, cmd)
 	if not cmd then
-		cmd = {
-			id = -uDefID,
-			name = UnitDefs[uDefID].name,
-			params = {}
-		}
+		cmd = { id = -uDefID, name = UnitDefs[uDefID].name, params = {} }
 	end
 	return cmd
 end
 
-
+---------------------------------------------------------------- main builders
 local function getGridForCategory(builderId, buildOptions, currentCategory)
-	local options = {}
-	-- lay out the category
-	if currentCategory then
-		local undefinedOpts = {}
-		local categoryIndex = getCategoryIndex(currentCategory)
-
-		-- first we go through all the buildopts that have defined positions and place them appropriately
-		for _, opt in pairs(buildOptions) do
-			if hasUnitGrid[builderId] and hasUnitGrid[builderId][opt] then
-				if unitGridPos[builderId][categoryIndex][opt] then
-					local row = string.sub(unitGridPos[builderId][categoryIndex][opt], 2, 2)
-					local col = string.sub(unitGridPos[builderId][categoryIndex][opt], 3, 3)
-					local index = col + ((row - 1) * columns)
-					if index then
-						options[index] = constructBuildOption(opt)
-					end
-				end
-			end
-			if not (hasUnitGrid[builderId] and hasUnitGrid[builderId][opt]) and unitCategories[opt] == currentCategory then
-
-				-- if this unit doesn't have a defined position in the grid, find an empty spot for it
-				table.insert(undefinedOpts, opt)
-			end
-		end
-
-		-- everything that doesn't have a defined position (i.e. scav units) gets placed into the grid in any empty spots.
-		for i = 1, 24 do
-			if #undefinedOpts < 1 then break end
-			if not options[i] then
-				local opt = table.remove(undefinedOpts, 1)
-				options[i] = constructBuildOption(opt)
+	local options, undefinedOpts = {}, {}
+	if not currentCategory then return options end
```

Comment:
Won't this cause every single build option to be paginated in one giant category? That is just the legacy build menu, but smaller.